static/app/utils/profiling/profile/sentrySampledProfile.tsx

```diff
@@ -0,0 +1,139 @@
+import {lastOfArray} from 'sentry/utils';
+import {CallTreeNode} from 'sentry/utils/profiling/callTreeNode';
+
+import {Frame} from './../frame';
+import {Profile} from './profile';
+import {createSentrySampleProfileFrameIndex} from './utils';
+
+// This is a simplified port of speedscope's profile with a few simplifications and some removed functionality.
+// head at commit e37f6fa7c38c110205e22081560b99cb89ce885e
+
+// We should try and remove these as we adopt our own profile format and only rely on the sampled format.
+export class SentrySampledProfile extends Profile {
+  static FromProfile(
+    samples: Profiling.SentrySampledProfile['profile']['samples'],
+    stacks: Profiling.SentrySampledProfile['profile']['stacks'],
+    frameIndex: ReturnType<typeof createSentrySampleProfileFrameIndex>
+  ): Profile {
+    const startedAt = parseInt(samples[0].elapsed_since_start_ns, 10);
+    const endedAt = parseInt(samples[samples.length - 1].elapsed_since_start_ns, 10);
+    const threadId = parseInt(samples[0].thread_id, 10);
+
+    const profile = new SentrySampledProfile({
+      duration: endedAt - startedAt,
+      startedAt,
+      endedAt,
+      unit: 'nanoseconds',
+      name: threadId.toString(),
```

Comment:
we should infer name from the transaction. Iirc each profile should be associated with at least one transaction. I would try and infer this profile.transactions[0].name as it should always be available. @phacops this may actually be a good case for us to add a profile title field or equivalent to the profile itself so we dont have to infer it from the transaction (e.g what do we do if there are multiple transactions with same name)?

I would also try and make this like a format of `profile name (thread)`. We should be able to pull thread name from thread_metadata[samples[0]]?.name. In the end, this would probably look something like `profilename (thread: main)`. In case name is not present, we should just fallback to `profilename (thread: 0)` where 0 is the threadid. I also wouldn't mind if we try and show both tid and thread name to avoid confusion

Reply:
Right now, since we guarantee 1 profile, 1 transaction, selecting the transaction name is the right move. I don't think this will ever change as for display, we only care to show a profile with 1 transaction in mind.